classes/invitation/core/Invitation.php

```diff
@@ -54,53 +94,105 @@ public function __construct(?InvitationModel $invitationModel = null)
         $this->fillFromPayload();
     }
 
-    public function initialize(?int $userId = null, ?int $contextId = null, ?string $email = null)
+    /**
+     * Function that adds the invitation to the database, initialising its main attributes.
+     * Either userId or email is taken into account - if both are defined, the userId is passed into the invitation.
+     * It removes all other InvitationStatus::INITIALIZED invitation from the database, that correspont to
+     * the same main attributes.
+     */
+    public function initialize(?int $userId = null, ?int $contextId = null, ?string $email = null, ?int $inviterId = null): void
     {
         if (!isset($userId) && !isset($email)) {
-            throw new Exception("Invitation should contain at least one user id or an invited email')");
+            throw new Exception("Invitation should contain the user id or an invited email.')");
+        }
+
+        $userIdUsed = null;
+        $emailUsed = null;
+        if (isset($userId)) {
+            $userIdUsed = $userId;
+        } else {
+            $emailUsed = $email;
         }
 
-        $this->invitationModel->userId = $userId;
+        InvitationModel::byStatus(InvitationStatus::INITIALIZED)
+            ->when($userIdUsed !== null, function ($query) use ($userIdUsed) {
```

Comment:
I prefer shorthand `fn()` syntax, and typehinted parameters. This would be e.g.:
```php
 ->when($userIdUsed !== null, fn (Builder $q) => $q->byUserId($userIdUsed))
```

Reply:
Changed.